dags/analyzer_helper/discourse/extract_raw_data.py

```diff
@@ -0,0 +1,212 @@
+from datetime import datetime
+from typing import Dict, List, Optional
+
+from analyzer_helper.discourse.utils.convert_date_time_formats import (
+    DateTimeFormatConverter,
+)
+from github.neo4j_storage.neo4j_connection import Neo4jConnection
+from hivemind_etl_helpers.src.utils.mongo import MongoSingleton
+
+
+class ExtractRawInfo:
+    def __init__(self, forum_endpoint: str, platform_id: str):
+        """
+        Initialize the ExtractRawInfo with the forum endpoint, platform id and set up Neo4j and MongoDB connection.
+        """
+        self.neo4jConnection = Neo4jConnection()
+        self.driver = self.neo4jConnection.connect_neo4j()
+        self.forum_endpoint = forum_endpoint
+        self.converter = DateTimeFormatConverter()
+        self.client = MongoSingleton.get_instance().client
+        self.platform_db = self.client[platform_id]
+        self.rawmemberactivities_collection = self.platform_db["rawmemberactivities"]
+
+    def close(self):
+        """
+        Close the Neo4j connection.
+        """
+        self.driver.close()
+
+    def fetch_post_details(
+        self, created_at: Optional[str] = None, comparison: Optional[str] = None
+    ) -> list:
+        """
+        Fetch details of posts from the Discourse forum.
+
+        :param created_at: Optional datetime string to filter posts created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of dictionaries containing post details.
+        """
+        if comparison:
+            assert comparison in {
+                "gt",
+                "gte",
+            }, "comparison must be either 'gt' or 'gte'"
+
+        where_clause = ""
+        if created_at and comparison:
+            operator = ">" if comparison == "gt" else ">="
+            where_clause = f"WHERE post.createdAt {operator} $createdAt"
+        query = f"""
+        MATCH (forum:DiscourseForum {{endpoint: $forum_endpoint}})
+        WITH forum
+        MATCH (topic:DiscourseTopic {{forumUuid: forum.uuid}})
+        MATCH (topic)-[:HAS_POST]->(post:DiscoursePost)
+        {where_clause}
+        OPTIONAL MATCH (post)<-[:POSTED]-(author:DiscourseUser)
+        OPTIONAL MATCH (post)<-[:LIKED]-(likedUser:DiscourseUser)
+        OPTIONAL MATCH (post)-[:REPLY_TO]->(repliedPost:DiscoursePost)
+        OPTIONAL MATCH (repliedPost)<-[:POSTED]-(repliedAuthor:DiscourseUser)
+        RETURN
+        post.id AS post_id,
+        author.id AS author_id,
+        post.createdAt AS created_at,
+        author.name AS author_name,
```

Comment:
We don't need author name. Please remove this from the return statement of the query